mass_mailing_list_dynamic/models/mail_mass_mailing_list.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2017 Tecnativa - Jairo Llopis
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import safe_eval
+
+
+class MassMailingList(models.Model):
+    _inherit = "mail.mass_mailing.list"
+
+    dynamic = fields.Boolean(
+        help="Set this list as dynamic, to make it autosynchronized with "
+             "partners from a given criteria.",
+    )
+    sync_method = fields.Selection(
+        [
+            ("add", "Only add new records"),
+            ("full", "Add and remove records as needed"),
+        ],
+        default="add",
+        required=True,
+        help="Choose the syncronization method for this list if you want to "
+             "make it dynamic",
+    )
+    sync_domain = fields.Char(
+        string="Synchronization critera",
+        default="[('opt_out', '=', False), ('email', '!=', False)]",
```

Comment:
```suggestion
        default="[('is_blacklisted', '=', False), ('email', '!=', False)]",
```

Reply:
By the way, there's no more opt out in res.partner, so all the related stuff should be adapted to the new blacklist system